staging/src/k8s.io/apiserver/pkg/endpoints/request/webhook_duration.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+func sumAgg(d1 time.Duration, d2 time.Duration) time.Duration {
+	return d1 + d2
+}
+
+func maxAgg(d1 time.Duration, d2 time.Duration) time.Duration {
+	if d1 > d2 {
+		return d1
+	}
+	return d2
+}
+
+type latencyTracker struct {
+	clock   clock.Clock
+	latency time.Duration
+	mu      *sync.RWMutex
+	agg     func(time.Duration, time.Duration) time.Duration
+}
+
+func (t *latencyTracker) Track(f func()) {
+	startedAt := t.clock.Now()
+	defer func() {
+		duration := t.clock.Since(startedAt)
+		t.mu.Lock()
+		defer t.mu.Unlock()
+		t.latency = t.agg(t.latency, duration)
+	}()
+
+	f()
+}
+
+func (t *latencyTracker) GetLatency() time.Duration {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	return t.latency
+}
+
+type WebhookDuration struct {
+	AdmitTimer    *latencyTracker
```

Comment:
Add LatencyTracker interface and make latencyTracker its implementation.

And use LatencyTracker here.

Reply:
Thanks, done.